### PR TITLE
Blacklist other user accounts

### PR DESCRIPTION
**Pull request by roschaefer**
See https://github.com/Human-Connection/WebApp/issues/211

### CODE REVIEW

File: features/step_definitions/steps.js [hunk 1]
```diff
@@ -109,3 +130,71 @@ When('you create your user settings via POST request to {string} with:', functio
...
+});
+
+When('you read through the comments of that post', function (callback) {
+  getRequest('/comments', callback);
+});
+
+Then('you will see a hint instead of a comment:', function (hint) {
+  const comment = httpResponse.data[0];
+  expect(comment.content).to.eq(hint);
+  expect(comment.contentExcerpt).to.eq(hint);
```
Comment (roschaefer) [inline features/step_definitions/steps.js]: @appinteractive @sebako86 I created three cucumber scenarios:
1. Write blacklist into `usersettings`
2. News feed should not include blacklisted usercontent
3. Comments should conceal blacklisted comments

Regarding 3, do you want the API response to include the actual comment content and the frontend displays a computed `visibleContent` or do you want the server to respond with the displayed content only?

As you can see in the implementation of the test, I am assuming the latter solution.
The other solution would enable a feature to unhide concealed user comments on the client side.

Comment (sebako86) [reply features/step_definitions/steps.js]: 1. & 2. cool
3. no need to have it handled on client side, let's safe the bandwidth for now and fetch any concealed content on demand.

**Comment by sebako86** [comment]
addition that just came to my mind: can you please also ensure that user with the role admin or moderator cannot be blocked by anyone? Thanks.

**Comment by roschaefer** [comment]
@sebako86 yup, and also not the user him-/herself

Shall I check these conditions on `service('usersettings').create` or on `service('contributions').find`? I prefer first option.

**Comment by sebako86** [comment]
@roschaefer I think on create is the better way to go, yes. :)

**Comment by Lulalaby** [comment]
@roschaefer is that ready?

**Comment by mimicc83** [comment]
@roschaefer looking all good to me, I am just wondering if we had a hook isAdmin || isModerator to use instead of 

`const unblacklistable = users.data.find((user) => {
      return (user.role === 'admin') || (user.role === 'moderator');
    });`

@appinteractive any remarks?

**Comment by roschaefer** [comment]
Hi all, this is **NOT** ready. We have a couple of branches on which this PR is based
`fix-docker-setup` => `feathers-vuex` => `blacklist-feature`

Furthermore, @mattwr18 removed the exlusive follow vs. block behaviour. At the moment I'm discussing the first ready-to-merge PR in discord with @appinteractive.

Sorry, in my first contribution to HC I wanted too much at once.

**Comment by roschaefer** [comment]
I think this is now ready to merge. @appinteractive merge before we make use of it in the UI?

**Comment by roschaefer** [comment]
ah, forgot one thing: Test in docker

**Comment by roschaefer** [comment]
Testing in docker seems fine, this is good to merge.

Strangely, it seems that seeder is also generating some seed data in `test` environment although the respective config is set to `false`. :thinking: nevertheless, testing works, so this can be merged.

**Pull request merged**